scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -2081,15 +2086,20 @@ class FormatOps(
     }
 
     object BlockImpl extends Impl {
+
       def tryGetSplits(ft: FormatToken, nft: FormatToken)(implicit
-          style: ScalafmtConfig
+          style: ScalafmtConfig,
+          line: sourcecode.Line
       ): Option[Seq[Split]] = {
         val leftOwner = ft.meta.leftOwner
         findTreeWithParentSimple(nft.meta.rightOwner)(_ eq leftOwner) match {
           case Some(t: Term.Block)
-              if t.stats.lengthCompare(1) > 0 && isBlockStart(t, nft) =>
+              if (t.stats.lengthCompare(1) > 0 ||
+                getBlockSingleStat(t).exists(_.isNot[Term]))
```

Comment:
is it isNot[term], or is[Defn]?

Reply:
I think that is safer, but Defn should suffice

Reply:
there's a similar logic in RedundantBraces (to avoid removing them), perhaps we should have a shared check.

Reply:
and if it's safer, let's keep it. you should know better anyway. i still haven't figured out the difference between tree and term.

Reply:
> there's a similar logic in RedundantBraces (to avoid removing them), perhaps we should have a shared check.

that one checks for Defn or Term.Function (`innerOk`). somewhat different, not much to share. but perhaps we should change `innerOk` to do `tree.isNot[Term] || tree.is[Term.Function]` (similar to this proposal).

Reply:
Changed the logic in RedundantBraces a bit too.